addons/survey/models/survey_question.py

```diff
@@ -264,6 +264,29 @@ def _compute_save_as_email(self):
             if question.question_type != 'char_box' or not question.validation_email:
                 question.save_as_email = False
 
+    @api.depends('survey_id.session_speed_rating_time_limit', 'is_time_limited')
+    def _compute_time_limit(self):
+        """If time limit value is set manually on the question, we do not update it
+        when changing the time limit value of the survey or disabling it.
+        """
+        for question in self:
+            if not question.is_time_limited:
+                question.time_limit = False
+            if not question.survey_id.session_speed_rating:
+                if question.time_limit == question.survey_id.session_speed_rating_time_limit:
+                    question.time_limit = False
+                    question.is_time_limited = False
+            elif question.is_time_limited:
+                if question.time_limit == 0 or question.time_limit == question._origin.survey_id.session_speed_rating_time_limit:
+                    question.time_limit = question.survey_id.session_speed_rating_time_limit or 0
+            else:
+                if question._origin.survey_id.session_speed_rating_time_limit == 0:
+                    question.time_limit = question.survey_id.session_speed_rating_time_limit or 0
+                    question.is_time_limited = True
```

Comment:
Isn't it possible to extract this into a '_compute_is_time_limited' and make it also computed editable ? It is not a good practice to have a standard field that is updated as a side-effect in a compute linked to another field... I feel like it is better to 1/ compute is_time_limited 2/ use it to compute time_limit (even if it duplicates a bit of code due to condition to reflect in both compute). What do you think ?